File: src/org/likexin/bfs/LevelOrder.java
package org.likexin.bfs;

import org.likexin.binarytree.TreeNode;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class LevelOrder {

  public List<List<Integer>> levelOrder(TreeNode root) {
    List<List<Integer>> results = new ArrayList<>();
    if (root == null) {
      return results;
    }

    Queue<TreeNode> queue = new LinkedList<>();
    queue.offer(root);

    while (!queue.isEmpty()) {
      List<Integer> list = new ArrayList<>();
      int size = queue.size();
      for (int i = 0; i < size; i++) {
        TreeNode node = queue.poll();
        if (node.left != null) {
          queue.offer(node.left);
        }
        if (node.right != null) {
          queue.offer(node.right);
        }
        list.add(node.val);
      }
      results.add(list);
    }
    return results;
  }

}

File: Python/bfs/LevelOrder.py
class Solution:
  """
    @param root: A Tree
    @return: Level order a list of lists of integer
  """
  def levelOrder(self, root):
    self.res = []
    if not root:
      return []
    q = [root]
    while q:
      temp = []
      self.res.append([n.val for n in q])
      for node in q:
        if node.left:
          temp.append(node.left)
        if node.right:
          temp.append(node.right)
      q = temp
    return self.res
      